src/renderer/services/ethereum/transaction.ts

```diff
@@ -8,22 +9,70 @@ import * as RxOp from 'rxjs/operators'
 
 import { LiveData } from '../../helpers/rx/liveData'
 import * as C from '../clients'
+import { ethRouterABI } from '../const'
 import { ApiError, ErrorId, TxHashLD } from '../wallet/types'
-import { ApproveParams, Client$, IsApprovedLD, TransactionService } from './types'
+import { ApproveParams, CallRouterParams, Client$, TransactionService, IsApprovedLD } from './types'
 
 export const createTransactionService = (client$: Client$): TransactionService => {
   const common = C.createTransactionService(client$)
 
+  const runCallRouterDeposit$ = (client: EthClient, params: CallRouterParams): TxHashLD =>
+    FP.pipe(
+      Rx.from(client.estimateGasPrices()),
+      RxOp.switchMap((gasPrices) => {
+        const amount = params.assetAddress === ETHAddress ? baseAmount(0) : params.amount
+        return Rx.from(
+          client.call<{ hash: TxHash }>(params.router, ethRouterABI, 'deposit', [
+            params.vault,
+            params.assetAddress,
+            amount.amount().toFixed(),
+            params.memo,
+            params.assetAddress === ETHAddress
+              ? {
+                  value: params.amount.amount().toFixed(),
```

Comment:
Please add a comment why we do need `toFixed` for amounts here and in next lines.